tests/testthat/test-available-packages.R

```diff
@@ -194,3 +182,20 @@ test_that("we can query the R universe", {
   expect_identical(lhs, rhs)
 
 })
+
+test_that("available_packages() tolerates missing repositories", {
+  skip_on_cran()
+  skip_if_not(grepl("\\bbinary\\b", .Platform$pkgType))
+
+  dbs <- available_packages(
```

Comment:
Add snapshot test?

Reply:
What should the snapshot capture? (since the set of available packages from CRAN will change over time)

Reply:
It's also possible that drat could start providing binaries in the future (or at least provide the infrastructure so `available.packages()` doesn't complain) so I think a snapshot test doesn't quite fit.

Reply:
That is a very good point! I'll update in a separate PR.